itests/common/src/main/java/org/wildfly/camel/test/common/utils/WildFlyCli.java

```diff
@@ -0,0 +1,183 @@
+            super();
+            this.command = command;
+            this.exitValue = exitValue;
+            this.stdOut = stdOut;
+            this.stdErr = stdErr;
+        }
+
+        /**
+         * @return this {@link WildFlyCliResult}
+         * @throws RuntimeException
+         *             in case {@link #exitValue} != 0 or {@link #stdErr} is not empty
+         */
+        public WildFlyCliResult assertSuccess() {
+            if (exitValue != 0) {
+                throw new RuntimeException(String.format("Command %s returned %d.\n\nstdout: %s\n\nstdErr: %s",
+                        Arrays.toString(command), exitValue, stdOut, stdErr));
+            }
+            if (!stdErr.isEmpty()) {
+                throw new RuntimeException(
+                        String.format("Command %s exited with non empty stdErr: %s", Arrays.toString(command), stdErr));
+            }
+            return this;
+        }
+
+        /**
+         * @return the numeric exit value returned by the interpreter running {@code jboss-cli}
+         */
+        public int getExitValue() {
+            return exitValue;
+        }
+
+        /**
+         * @return the text the {@code jboss-cli} process has written to {@code stderr} or an empty string if no text
+         *         was writtent to {@code stderr}
+         */
+        public String getStdErr() {
+            return stdErr;
+        }
+
+        /**
+         * @return the text the {@code jboss-cli} process has written to {@code stdout} or an empty string if no text
+         *         was writtent to {@code stdout}
+         */
+        public String getStdOut() {
+            return stdOut;
+        }
+    }
+
+    private final Path wildFlyHome;
+
+    public WildFlyCli(Path wildFlyHome) {
+        super();
+        this.wildFlyHome = wildFlyHome;
+    }
+
+    /**
+     * Run the given {@code cliScript} using {@code jboss-cli.[sh|bat]} and the appropriate current platform's script
+     * interpreter.
+     *
+     * @param cliScript
+     *            the {@link Path} to the script to run
+     * @return a WildFlyCliResult
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    public WildFlyCliResult run(Path cliScript) throws IOException, InterruptedException {
+        final ProcessBuilder pb = new ProcessBuilder();
+        final String[] command;
+        if (System.getProperty("os.name").toLowerCase().contains("win")) {
+            final String jbossCliPath = wildFlyHome.resolve("bin/jboss-cli.bat").normalize().toString();
+            command = new String[] { "cmd.exe", jbossCliPath, "--connect", "--echo-command",
+                    "--file=" + cliScript.normalize().toString() };
+        } else {
+            final String jbossCliPath = wildFlyHome.resolve("bin/jboss-cli.sh").normalize().toString();
+            command = new String[] { "/bin/sh", jbossCliPath, "--connect", "--echo-command",
```

Comment:
Do we need to run with `/bin/bash` (or cmd.exe). The script should always be executable and it has the relevant shebang line for the required interpreter at the beginning of the file.

Reply:
Yes, perhaps not `/bin/bash` but quite certainly `cmd.exe`. Let me check.

Reply:
I was not able to set up a working Windows CI for the current branch of WFC, but I found I did it in the past without `cmd.exe` in code [1] that certainly was tested on Windows. So you are right, thanks!

[1] https://github.com/srcdeps/srcdeps-core/blob/master/srcdeps-core/src/main/java/org/srcdeps/core/impl/builder/GradlewBuilder.java#L40